Source/SharpNeedle/HedgehogEngine/Mirage/SplinePath.cs

```diff
@@ -0,0 +1,283 @@
+        Right = reader.Read<Vector3>();
+    }
+
+    public void Write(BinaryObjectWriter writer)
+    {
+        writer.Write(Left);
+        writer.Write(Right);
+    }
+}
+
+public class UserData : IBinarySerializable
+{
+    public string Name { get; set; }
+    public DataType Type { get; set; }
+    public uint Value { get; set; }
+    public string? StringValue { get; set; }
+
+    public unsafe float FloatValue
+    {
+        get
+        {
+            var v = Value;
+            return *(float*)&v;
+        }
+
+        set => Value = *(uint*)&value;
+    }
+
+    public void Read(BinaryObjectReader reader)
+    {
+        Name = reader.ReadStringOffset();
+
+        Type = reader.Read<DataType>();
+        switch (Type)
+        {
+            case DataType.Float:
+            case DataType.Integer:
+                Value = reader.Read<uint>();
+                break;
+
+            case DataType.String:
+                StringValue = reader.ReadStringOffset();
+                break;
+
+            default:
+                throw new NotImplementedException($"{Type} is not recognized");
+        }
+    }
+
+    public void Write(BinaryObjectWriter writer)
+    {
+        writer.WriteStringOffset(StringBinaryFormat.NullTerminated, Name);
+
+        writer.Write(Type);
+        switch (Type)
+        {
+            case DataType.Integer:
+            case DataType.Float:
+                writer.Write(Value);
+                break;
+
+            case DataType.String:
+                writer.WriteStringOffset(StringBinaryFormat.NullTerminated, StringValue);
+                break;
+
+            default:
+                throw new NotImplementedException($"{Type} is not recognized");
+        }
+    }
+
+    public override string ToString()
+    {
+        return Name;
+    }
+
+    public enum DataType
+    {
+        Integer,
+        Float, // Guessed
+        String
+    }
+}
+
+public class UnknownStruct : IBinarySerializable
+{
+    public struct SubUnknown1 : IBinarySerializable
+    {
+        public uint Field00 { get; set; }
+        public float Field04 { get; set; }
+
+        public void Read(BinaryObjectReader reader)
+        {
+            Field00 = reader.Read<uint>();
+            Field04 = reader.Read<float>();
+        }
+
+        public void Write(BinaryObjectWriter writer)
+        {
+            writer.Write(Field00);
+            writer.Write(Field04);
+        }
+    }
+
+    public struct SubUnknown2 : IBinarySerializable
+    {
+        public int Field00 { get; set; }
+        public int Field04 { get; set; }
+
+        public void Read(BinaryObjectReader reader)
+        {
+            Field00 = reader.Read<int>();
+            Field04 = reader.Read<int>();
+        }
+
+        public void Write(BinaryObjectWriter writer)
+        {
+            writer.Write(Field00);
+            writer.Write(Field04);
+        }
+    }
+
+    public struct SubUnknown3 : IBinarySerializable
```

Comment:
Maybe remove this and just use a `List<int>` since this is a single value and likely not a whole struct.

Reply:
Resolved in 6a814e4.